stumpy/stump.py

```diff
@@ -454,6 +471,10 @@ def stump(T_A, m, T_B=None, ignore_trivial=True):
         start, T_A, T_B, m, excl_zone, M_T, Σ_T, ignore_trivial
     )
 
+    T_B[
+        np.isnan(T_B)
+    ] = 0  # Remove all nan values from T_B only after first profile is calculated
```

Comment:
Because the distance profile for the first sub-sequence is computed using core.mass. If the first sub-sequence contains a `np.nan` value and we replace it, mass has no way of knowing this.